.github/workflows/docker-openmetadata-server.yml

```diff
@@ -9,48 +9,146 @@
+          mvn versions:set -DnewVersion=0.13.0
+          mvn -DskipTests clean install
+
+      - name: Upload openmetadata application
+        uses: actions/upload-artifact@v3
+        with:
+          name: openmetadata-binary
+          path: /home/runner/work/OpenMetadata/OpenMetadata/openmetadata-dist/target/*.tar.gz
+ 
+  release-project-event-release:
+    if: github.event_name == 'release'
+    name: Release app with release event
+    runs-on: ubuntu-latest
+    needs: maven-build
+    steps:
+      - name: Check trigger type
+        id: check_trigger
+        run: echo "GITHUB_RELEASE_TAG=0.13.0" >> $GITHUB_OUTPUT
+        
+      - name: Download application
+        uses: actions/download-artifact@v2
+        with:
+          name: openmetadata-binary
+
+      - name: Get release
+        id: get_release
+        uses: bruceadams/get-release@v1.3.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Upload app tar to Github Release
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          upload_url: ${{ steps.get_release.outputs.upload_url }}
+          asset_path: ./openmetadata-${{ steps.check_trigger.outputs.GITHUB_RELEASE_TAG }}.tar.gz
+          asset_name: openmetadata-${{ steps.check_trigger.outputs.GITHUB_RELEASE_TAG }}.tar.gz
+          asset_content_type: application/tar.gz
+          
+  release-project-event-workflow_dispatch:
+    if: github.event_name == 'workflow_dispatch'
+    name: Release app with workflow_dispatch event
+    runs-on: ubuntu-latest
+    needs: maven-build
+    steps:
+
+      - name: Attach to Release
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        id: attach_release
+        run: |
+          echo "UPLOAD_URL=$(curl -H "Accept: application/vnd.github+json" -H "Authorization: Bearer $GITHUB_TOKEN" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/open-metadata/OpenMetadata/releases/tags/${{ inputs.GITHUB_RELEASE_TAG }}  | jq .upload_url | tr -d '"' )" >> $GITHUB_OUTPUT
+
+      - name: Download application
+        uses: actions/download-artifact@v2
+        with:
+          name: openmetadata-binary
+
+      - name: Upload release asset
+        uses: actions/upload-release-asset@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          asset_path: ./openmetadata-${{ inputs.GITHUB_RELEASE_TAG }}.tar.gz
+          upload_url: ${{ steps.attach_release.outputs.UPLOAD_URL }}
+          asset_name: openmetadata-${{ inputs.GITHUB_RELEASE_TAG }}.tar.gz
+          asset_content_type: application/tar.gz
+
   push_to_docker_hub:
     runs-on: ubuntu-latest
     env:
-      input: ${{ github.event.inputs.tag }}
-      
+      DOCKER_RELEASE_TAG: ${{ github.event.inputs.DOCKER_RELEASE_TAG }}
+    needs: [maven-build]
```

Comment:
This workflow should wait for file upload first to Github release and then start. Or else for release scenario, this will run in parallel with  `release-project-event-release` or `release-project-event-workflow_dispatch` job and will not get the updated binary required for building the docker image.